department-of-veterans-affairs/vets-website
# 98144 Sign-in widget: Update verify alert to use service-specific buttons and messaging

## Are you removing, renaming or moving a folder in this PR?

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary
- Change the verify alert to use service-specific buttons that is relevant to the CSP user signed in with
- Update verify alert messaging 

## Related issue(s)

- department-of-veterans-affairs/va.gov-team/issues/98144
- department-of-veterans-affairs/va.gov-team/issues/97279

## Testing done

- Add/update unit tests
- Manual testing in local development:
  - 'yarn watch` from `vets-website`
  - `foreman start -m all=1,clamd=0,freshclam=0` from `vets-api`
  - Visit `http://localhost:3001/sign-in/mocked-auth` to use mocked authentication for Credential Service Provider:
    - ID.me  (profile: joe.niquette+idmetest5@oddball.io | MPI not mocked)
    - Login.gov (profile: joe.niquette+lgovial1@oddball.io | MPI not mocked)
    - no existing test users for My HealtheVet, so manual code change to view the SignInOtherAccount for MHV Basic account)
    - Visit `http://localhost:3001/health-care/schedule-view-va-appointments/` to check that relevant alert is rendered

## Screenshots




|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    <img width="383" alt="Screenshot 2024-12-12 at 10 11 43 AM" src="https://github.com/user-attachments/assets/9dbd4de6-7c79-44cc-a5a7-e19dafc62970" />    |   <img width="383" alt="Screenshot 2024-12-12 at 10 05 58 AM" src="https://github.com/user-attachments/assets/971728b6-37f7-400a-9fe0-925af8db479a" />    |
| Desktop  |    <img width="687" alt="Screenshot 2024-12-12 at 10 13 38 AM" src="https://github.com/user-attachments/assets/ce05fb09-8fe5-4d67-86da-6336bc4bc9d9" />   |   <img width="673" alt="Screenshot 2024-12-10 at 9 02 01 PM" src="https://github.com/user-attachments/assets/5da03fd2-3fa9-4faa-808f-b8786ab77d06" />   |
| Mobile  |   <img width="383" alt="Screenshot 2024-12-12 at 10 11 43 AM" src="https://github.com/user-attachments/assets/9dbd4de6-7c79-44cc-a5a7-e19dafc62970" />     |    <img width="383" alt="Screenshot 2024-12-12 at 10 05 26 AM" src="https://github.com/user-attachments/assets/8bf254c7-c531-4ba6-a484-41eeb920c7e2" />   |
| Desktop |   <img width="687" alt="Screenshot 2024-12-12 at 10 13 38 AM" src="https://github.com/user-attachments/assets/ce05fb09-8fe5-4d67-86da-6336bc4bc9d9" />     |  <img width="673" alt="Screenshot 2024-12-10 at 9 03 47 PM" src="https://github.com/user-attachments/assets/540f6c02-012b-4621-b98f-0341631bcd0f" />  |
| Mobile |   <img width="383" alt="Screenshot 2024-12-12 at 10 11 43 AM" src="https://github.com/user-attachments/assets/9dbd4de6-7c79-44cc-a5a7-e19dafc62970" />      |   <img width="386" alt="Screenshot 2024-12-16 at 2 05 18 PM" src="https://github.com/user-attachments/assets/8460f2ad-351a-46e6-9f61-6ea9af6b5c0a" />  |
| Desktop |   <img width="687" alt="Screenshot 2024-12-12 at 10 13 38 AM" src="https://github.com/user-attachments/assets/2b93f250-ffbf-4d16-9344-beca5c7459ba" />     |   <img width="714" alt="Screenshot 2024-12-16 at 2 04 27 PM" src="https://github.com/user-attachments/assets/20a4e9a4-5692-4a15-8e0f-d6e7aefec312" />   |

## What areas of the site does it impact?

These changes impact _widgets_ that get embedded on static pages. The following are some paths that embed widgets that used this alert.

- `/health-care/schedule-view-va-appointments/`
- `/health-care/refill-track-prescriptions/`
- `/health-care/secure-messaging/`
- `/health-care/view-test-and-lab-results/`
- `health-care/get-medical-records/`

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
